src/encoded/audit/experiment.py

```diff
@@ -1746,6 +1748,65 @@ def audit_experiment_biosample_characterization(value, system, excluded_types):
             status = characterization.get('review', {}).get('status')
             sample_characterization_status.setdefault(status, [])
             sample_characterization_status[status].append(biosample['@id'])
+        # Check immediate parent of biosample for characterizations, via part_of or originated_from
+        # The parent biosample must match the child ontology, treatments, and applied modifications
+        if not sample_characterization_status:
+            child_mods = sorted(mods)
+            child_treatments = sorted(biosample.get('treatments', []))
+            relationship = ''
+            part_of_biosample = biosample.get('part_of', [])
+            originated_from_biosample = biosample.get('originated_from', [])
+            if (
+                part_of_biosample
+                and part_of_biosample.get('characterizations')
+            ):
+                part_ontology = part_of_biosample.get('biosample_ontology')
+                part_mods = sorted(part_of_biosample.get('applied_modifications', []))
+                part_treatments = sorted(part_of_biosample.get('treatments', []))
+                if (
+                    part_ontology == ontology
+                    and part_mods == child_mods
+                    and part_treatments == child_treatments
+                ):
+                    relationship = 'part_of'
+            elif (
+                originated_from_biosample
+                and originated_from_biosample.get('characterizations')
+            ):
+                orig_ontology = originated_from_biosample.get('biosample_ontology')
+                orig_mods = sorted(originated_from_biosample.get('applied_modifications', []))
+                orig_treatments = sorted(originated_from_biosample.get('treatments', []))
+                if (
+                    orig_ontology == ontology
+                    and orig_mods == child_mods
+                    and orig_treatments == child_treatments
+                ):
+                    relationship = 'originated_from'
+            if relationship != '':
+                parent_characterization_status = {
+                    c.get('review', {}).get('status')
+                    for c in biosample.get(relationship).get('characterizations')
```

Comment:
this is going to fail if biosample['relationship']['characterizations'] is not an iterable.  Maybe you secured that already?